syntax = "proto3";

package simRPC;

// Interface exported by the server.
service simRPC {
    rpc RecordMetrics(Metrics) returns (Metrics) {}
}

// 测试
// 返回四个关键值的str
message Metrics {
    Clock   clock = 1;
    Nodes   nodes = 2;
    Pods    pods = 3;
    Queue   queue = 4;
}

message Clock {
    string clock_Key = 1;
}

message Nodes {
    string nodes_metrics_key = 1;
}

message Pods {
    string pods_metrics_key = 1;
}

message Queue {
    string queue_metrics_key = 1;
}

message Result {
    int32 result = 1;
}